### PR TITLE
Master

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        cardList = new ArrayList<>();
+    }
+
+    private Deck(ArrayList<Card> list) {
+        cardList = (list != null) ? new ArrayList<>(list) : null;
+    }
+
+    public void generate(int pn) {
+        if (playersNumber == 0) {
+            playersNumber = (pn > 1 && pn < 5) ? pn : 2;    // TODO: определить дефолтное значение количества игроков
```
Comment (ZoXaL) [inline backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java]: Players number doesn't reflect with deck generation

File: backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+            Card.Color[] colors = Card.Color.values();
+            for (int i = 0; i < playersNumber * DEFAULT_CARDS_AMOUNT + 1; i++) {
+                cardList.add(new Card(rnd.nextInt(10) + 1, rnd.nextInt(3) + 1, colors[rnd.nextInt(3)]));
+            }
+        }
+    }
+
+    public void shuffle() {
```
Comment (ZoXaL) [inline backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java]: There is suitable JDK method: https://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#shuffle(java.util.List,%20java.util.Random)

File: backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            return null;
+        }
+    }
+
+    public Deck pullCards(int asDeck) {
+        return new Deck(pullCards());
+    }
+
+    public Card getStartCard() {
+        if (cardList!=null && !cardList.isEmpty() && cardList.size()%2 != 0)
```
Comment (ZoXaL) [inline backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java]: * `cardList` was initialized at constructor
* I think, it's ok just to peek a card without deletion it from deck

File: backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        }
+    }
+
+    public void shuffle() {
+        if (cardList!=null && !cardList.isEmpty())
+        for (int i = 0; i < playersNumber * DEFAULT_CARDS_AMOUNT; i++)
+            Collections.swap(cardList, rnd.nextInt(cardList.size()), rnd.nextInt(cardList.size()));
+    }
+
+    public ArrayList<Card> pullCards() {
```
Comment (ZoXaL) [inline backend/src/main/java/com/yatty/sevennine/backend/model/Deck.java]: `cardList` was initialized at constructor
P.S. It's bad practice to return null. Throw an exception or return empty list instead.

**Review by mikhail-chystsiakou — APPROVED** [review]

**Pull request merged**